docs/advanced/custom-map-controls.md

```diff
@@ -76,10 +58,115 @@ Additionally, event objects for different event types contain a subset of the fo
 
 ##### `handleEvent(event)`
 
-Called by `InteractiveMap` to handle pointer events.
+Called by the event manager to handle pointer events. This method delegate to the following methods to handle the default events:
+- `_onPanStart(event)`
+- `_onPan(event)`
+- `_onPanEnd(event)`
+- `_onPinchStart(event)`
+- `_onPinch(event)`
+- `_onPinchEnd(event)`
+- `_onDoubleTap(event)`
+- `_onWheel(event)`
+- `_onKeyDown(event)`
+
+##### `getMapState(overrides)`
+
+Get a new descriptor object of the map state. If specified, props in the `overrides` object override the current map props.
 
 ##### `setOptions(options)`
 
-Called by `InteractiveMap` when props change.
+Add/remove event listeners based on the latest `InteractiveMap` props.
+
+##### `setState(newState)`
+
+Save a persistent state (e.g. isDragging) for future use.
+
+##### `updateViewport(newMapState, extraProps, extraState)`
+
+Invoke `onViewportChange` callback with a new map state.
 
 
+### Source
+[map-controls.js](https://github.com/uber/react-map-gl/tree/3.0-release/src/utils/map-controls.js)
+
+
+## Examples
```

Comment:
Just as comparison: 
* In other frameworks, like luma.gl, the class reference docs often start with a Usage section. 
* I'm fine if different frameworks organize things a little differently, just pointing it out.